src/Task/Git/CommitMessage.php

```diff
@@ -240,17 +262,35 @@ private function getSpecialPrefixLength($string)
      *
      * @return bool
      */
-    private function subjectHasTrailingPeriod(ContextInterface $context)
+    private function subjectHasPunctuations(ContextInterface $context)
     {
-        $commitMessage = $context->getCommitMessage();
+        $subjectLine = $this->getSubjectLine($context);
 
-        if (trim($commitMessage) === '') {
+        if (trim($subjectLine) === '') {
             return false;
         }
 
-        $lines = $this->getCommitMessageLinesWithoutComments($commitMessage);
+        if (Str::containsOneOf($subjectLine, ['.', '!', '?', ','])) {
```

Comment:
`return Str::containsOneOf($subjectLine, ['.', '!', '?', ',']);`

Reply:
Good catch!

Reply:
This should resolve this conversation https://github.com/phpro/grumphp/pull/541/commits/8ee9e9895f1886a6397d44cdc7902fb8d1d8f00c